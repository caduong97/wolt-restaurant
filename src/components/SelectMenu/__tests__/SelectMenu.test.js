import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { SelectMenu } from '..'

describe('<SelectMenu />', () => {
  const mockSortAsc = jest.fn()
  const mockSortDesc = jest.fn()

  it('renders and matches snapshot', () => {
    const { container } = render(
      <SelectMenu sortAsc={mockSortAsc} sortDesc={mockSortDesc} />,
    )
    expect(container).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
  it('renders and matches snapshot', () => {
    const { getByTestId } = render(
      <SelectMenu sortAsc={mockSortAsc} sortDesc={mockSortDesc} />,
    )

    const sortAscButton = getByTestId('sort-ascending')
    const sortDescButton = getByTestId('sort-descending')

    fireEvent.click(sortAscButton)
    expect(mockSortAsc).toHaveBeenCalled()
    fireEvent.click(sortDescButton)
    expect(mockSortDesc).toHaveBeenCalled()
  })
})
