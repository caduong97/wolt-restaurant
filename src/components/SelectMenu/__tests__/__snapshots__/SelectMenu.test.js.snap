// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectMenu /> renders and matches snapshot 1`] = `
<div>
  <div
    class="select"
  >
    <button
      class="select__current"
      id="sort-button"
      type="button"
    >
      Sort name alphabetically
    </button>
    <div
      class="select__options"
    >
      <button
        class="select__options__item"
        data-testid="sort-ascending"
        type="button"
      >
        From A to Z
      </button>
      <button
        class="select__options__item"
        data-testid="sort-descending"
        type="button"
      >
        From Z to A
      </button>
    </div>
  </div>
</div>
`;
